web/src/app/services/app-state.service.ts

```diff
@@ -0,0 +1,22 @@
+import { Injectable } from '@angular/core';
+
+@Injectable()
+export class AppStateService {
```

Comment:
`app-state` sounds very redux like, which can become an anti-pattern. Can we have an  `authService` that keeps the state of the authentication?

Reply:
Okay will move this functionality to Auth Service.

Reply:
There is authentication service, can we use that?

Reply:
Yes @eddiejaoude I have changed the flow to use the firebase auth state check for auth guard